cores/arduino/WString.cpp

```diff
@@ -492,7 +492,13 @@ int String::compareTo(const String &s) const
 		if (buffer && len > 0) return *(unsigned char *)buffer;
 		return 0;
 	}
-	return strcmp(buffer, s.buffer);
+	const char *p1 = buffer;
+	const char *p2 = s.buffer;
+
+	while (*p1 == *p2++)
+		if ('\0' == *p1++)
+			return 0;
+	return (*(const char *)p1 - *(const char *)(p2 - 1));
```

Comment:
Oh yeah you're right, there are more functional changes that I missed. OK, so @bokibi can you please remove functional changes from this PR? leave header changes only, like the title suggests.

That said, I believe your "strcmp" re-implementation is a valid fix, and I would like to merge it also if we cannot fix the toolchain. Please make a new pull request for each functional change.